docs/pages/ESSupport.md

```diff
@@ -0,0 +1,368 @@
+			metadata: {
+				properties: {
+					"templateName": {
+						"type": "string",
+						"defaultValue": "sap.fe.templates.Page.view.Page"
+					}
+				},
+				"manifest": "json"
+			}
+		}
+));
+```
+
+**⛔️ Not supported: Object Expression**
+```javascript
+sap.ui.define(["sap/fe/core/TemplateAssembler"],
+	(TemplateAssembler) => {
+		const key =  "templateName";
+		return TemplateAssembler.getTemplateComponent(getMethods,
+		"sap.fe.templates.Page.Component", {
+			metadata: {
+				properties: {
+					[key]: {
+						"type": "string",
+						"defaultValue": "sap.fe.templates.Page.view.Page"
+					}
+				},
+				"manifest": "json"
+			}
+		})
+	}
+);
+```
+
+**⛔️ Not supported: Spread Element**
+```javascript
+sap.ui.define(["sap/fe/core/TemplateAssembler"],
+	(TemplateAssembler) => {
+		const myTemplate = {
+			templateName: {
+				type: "string",
+				defaultValue: "sap.fe.templates.Page.view.Page"
+			}
+		};
+		return TemplateAssembler.getTemplateComponent(getMethods,
+		"sap.fe.templates.Page.Component", {
+			metadata: {
+				properties: {
+					...myTemplate
+				},
+				"manifest": "json"
+			}
+		}
+)});
+```
+
+**⛔️ Not supported: Template Literal with Expression**
+```javascript
+sap.ui.define(["sap/fe/core/TemplateAssembler"],
+	(TemplateAssembler) => {
+		const name = "Component";
+		return TemplateAssembler.getTemplateComponent(getMethods,
+		`sap.fe.templates.Page.${name}`, {
+			metadata: {
+				properties: {
+					"templateName": {
+						"type": "string",
+						"defaultValue": "sap.fe.templates.Page.view.Page"
+					}
+				},
+				"manifest": "json"
+			}
+		}
+)});
+```
+
+## Library
+For libraries, some special cases require following best practices.
+
+### sap/ui/core/Core#initLibrary: Usage of Object Expression, Spread Element, and Template Literal
+Typically in the `library.js` of your library, the library is initialized. The object which is given to the `sap/ui/core/Core#initLibrary` call must not include any `Object Expression`, `Spread Element`, or `Template Literal` with expression.
+
+**✅ Best Practice:**
+```javascript
+sap.ui.getCore().initLibrary({
+	name: "my.lib"
+```
+
+**⛔️ Not supported: Object Expression**
+```javascript
+const mylib = {
+	key: "name"
+};
+sap.ui.getCore().initLibrary({
+	[mylib.key] : "my.lib",
+```
+
+**⛔️ Not supported: Spread Element**
+```javascript
+const mylib = {
+	name: "my.lib"
+};
+sap.ui.getCore().initLibrary({
+	...mylib
+```
+
+**⛔️ Not supported: Template Literal with Expression**
+```javascript
+const name = "my"
+sap.ui.getCore().initLibrary({
+	name: `{$name}.lib`
+```
+
+### JSDoc
+For libraries, the JSDoc output might be relevant. Therefore, please follow the below best practices.
+
+#### sap.ui.define/sap.ui.require: with Arrow function expression
+The JSDoc block of the module has to be written above the arrow function and not above the `sap.ui.define/sap.ui.require` command.
+
+**✅ Best Practice:**
+
+```javascript
+sap.ui.define(["Bar"],
+	/**
+	 * JSDoc block here
```

Comment:
to be checked if it works

Reply:
Verified with https://github.com/SAP/ui5-builder/blob/main/test/fixtures/library.j/main/src/library/j/SubControl.js#L12-L28 and https://github.com/SAP/ui5-builder/blob/main/test/expected/build/library.j/dest/test-resources/library/j/designtime/api.json#L85